pkg/csi/cinder/controllerserver.go

```diff
@@ -110,31 +112,56 @@ func (cs *controllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
 	}
 	content := req.GetVolumeContentSource()
 	var snapshotID string
-	var sourcevolID string
+	var sourceVolID string
+	var sourceBackupID string
 
 	if content != nil && content.GetSnapshot() != nil {
 		snapshotID = content.GetSnapshot().GetSnapshotId()
-		_, err := cloud.GetSnapshotByID(snapshotID)
-		if err != nil {
-			if cpoerrors.IsNotFound(err) {
-				return nil, status.Errorf(codes.NotFound, "VolumeContentSource Snapshot %s not found", snapshotID)
+
+		snap, snapErr := cs.Cloud.GetSnapshotByID(snapshotID)
+		if snapErr != nil && !cpoerrors.IsNotFound(snapErr) {
+			return nil, snapErr
+		}
+		// If the snapshot exists but is not yet available, fail.
+		if snapErr == nil && snap.Status != "available" {
```

Comment:
Since the check is 
`if snapErr == nil && snap.Status != "available" {...`
It first checks for the error to be `nil` (so snap exists) and only then would it check for the snap status, so we are protected :)